feat: Implement `n_distinct()` as macro with support for `na.rm = TRUE`

joakimlinde opened the pull request:
I wish it was possible to support multiple arguments to `n_distinct()` but I haven't been able to find a solution using macros. If you have a suggestion, please let me know. 

Fixes #572

joakimlinde committed:
feat: Implement n_distinct() as macro with support for na.rm = TRUE

krlmlr reviewed (commented):
Thanks, I like the approach. Can we add a couple of generated tests too (in the "mutate" part)?

krlmlr commented:
Also, the non-support of `n_distinct(na.rm = TRUE)` is documented in `limits.Rmd`, we should update that.

joakimlinde commented:
Sure, I'll take a look at that.

joakimlinde commented:
I've added mutate error message, test of this error message, and updated limits.R. Also beefed up the tests because I realized that the dplyr n-distinct tests don't run on the duckplyr version so I can't rely on them.

joakimlinde commented:
Forgot I can't use the new pipe. Fixed.

krlmlr commented:
Thank you very much!
